### PR TITLE
#174797543 CRUD on Bookings

### PR DESCRIPTION
**Pull request by jpirumvaa**
### What does this PR do?

1. Allow users to book an accomodation
2. Allow users to get all bookings
3. Allow users to get a single booking by id
4. Allow users to update a booking using a id
5. Allow users to delete a booking using its id
6. Allow users to delete all bookings at once

### Description of Task to be completed?

1. Pull the repo to get the project's latest version
2. Create a branch to work on
3. Select accommodation from the database.
4. Change` isBooked`  to be true and `userId` to be the id of a user who booked accommodation.
5. Write the whole CRUD
6. Validate User's Input
7. Write Tests
8. Add, commit, and push changes to GitHub

### How should this be manually tested?

1. Clone a repo, run `cd script-heroes-bn-backend` to go to the project's repo
2.  Run `npm install` to install all project dependencies then run `npm run dev` to run the project. 
3. Go to postman, then use the URL: `http://localhost:PORTNO/api/accomodations/book` . Make sure that you have `accomodationId` , `checkInDate`, `checkOutDate`, and `roomId` in the request's body.
4. Complete all `CRUD` operations

### Any background context you want to provide?

The request's body should look like this:
`{
 "accommodationId":2,
"roomId": 3,
    "checkInDate": "January 1, 2020",
    "checkOutDate": "Feb 2, 2020"
}`
### What are the relevant pivotal tracker stories?
[#174797543](https://www.pivotaltracker.com/n/projects/2465784)

### CODE REVIEW

**Review by dushimeemma — APPROVED** [review]
Put the exact endpoints in task description and put the allowed space on locale files otherwise smells good to me

**Review by valens10 — APPROVED** [review]

File: src/controllers/bookAccomodation.js [hunk 1]
```diff
@@ -1,49 +1,107 @@
...
-        if (!accomodation) {
-            return res.status(404).json({ message: res.__('No accomodation found') });
-        }
+  getBookings: async (req, res) => {
+    const bookings = await Bookings.findAll({
+      include: [
+        { model: Room, include: [{ model: Accomodation }] },
+        { model: User }
+      ]
+    });
```
Comment (Adadie) [inline src/controllers/bookAccomodation.js:10]: May you add functionality for only admins to get all bookings, I think it wouldn't be good for everyone to get all this info

File: src/controllers/bookAccomodation.js [hunk 2]
```diff
@@ -1,49 +1,107 @@
...
+    if (!deletedBooking) {
+      return res
+        .status(404)
+        .json({ message: res.__('No booking with specified id') });
+    }
+    return res
+      .status(200)
+      .json({ message: res.__('Booking deleted successfully') });
+  },
```
Comment (Adadie) [inline src/controllers/bookAccomodation.js]: Try to see if users logged can get, edit and delete their own bookings

**Review by Adadie — COMMENTED** [review]
I think you've created another location migration, check if it's the same as the one on develop

File: src/database/migrations/20201008113305-create-location.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  // eslint-disable-next-line no-unused-vars
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.dropTable('Locations');
+  }
+};
```
Comment (Adadie) [inline src/database/migrations/20201008113305-create-location.js]: Check if this new migration is necessarry

**Review by b0nbon1 — APPROVED** [review]

**Comment by b0nbon1** [comment]
Please resolve the conflicts and It'll be good to go

**Review by Adadie — APPROVED** [review]

**Review by Adadie — APPROVED** [review]

**Pull request merged**